src/server/game/Battlegrounds/Battleground.cpp

```diff
@@ -919,6 +922,32 @@ void Battleground::EndBattleground(PvPTeamId winnerTeamId)
     sScriptMgr->OnBattlegroundEnd(this, GetTeamId(winnerTeamId));
 }
 
+bool Battleground::SpiritofCompetitionEvent(PvPTeamId winnerTeamId)
+{
+    for (auto const& [playerGuid, player] : m_Players)
+    {
+        TeamId bgTeamId = player->GetBgTeamId();
+        if (bgTeamId == GetTeamId(winnerTeamId))
```

Comment:
for self: Remove this if, it is wrong, all players should be eligible for the tabard